standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -3101,6 +3100,22 @@ public boolean dropPartition(String catName, String dbName, String tableName,
     return success;
   }
 
+  @Override
+  public boolean dropPartition(String catName, String dbName, String tableName, String partName)
+      throws MetaException, NoSuchObjectException, InvalidObjectException, InvalidInputException {
+    boolean success = false;
+    try {
+      openTransaction();
+      dropPartitionsInternal(catName, dbName, tableName, Arrays.asList(partName), true, true);
```

Comment:
I don't think this would improve the performance by any means. Consider dropping 10k partitions, each partition would have to access same number of tables in the underlying db to update the records, so it makes sense to batch them and implement with direct SQL. But for a single partition since it'll access same number of tables in the DB, I don't think it'll make sense to implement this feature.
For example, [HIVE-26035](https://issues.apache.org/jira/browse/HIVE-26035) (see the details in the jira) proved that implementing direct SQL actually improved the performance by running against benchmark tests.
Similarly can you provide any evidence that this patch also has an edge by running those tests? (Probably you might have to add some tests(e.g: dropping 10K+ single partitions).

Reply:
cc @VenuReddy2103

Reply:
We have some test for direct sql and jdo sql before, it shows direct sql is always faster even for simple operation like get connection. Let me add the benchmark for dropping multiple single partitions.